core/templates/domain/question/question-update.service.ts

```diff
@@ -91,8 +91,10 @@ export class QuestionUpdateService {
           // All rules input types which are translatable are subclasses of
           // BaseTranslatableObject having dict structure with contentId
           // as a key.
-          if (ruleInput && 'contentId' in ruleInput) {
-            allContentIdsSet.add(ruleInput.contentId);
+          if (ruleInput.hasOwnProperty('contentId')) {
+            if (ruleInput && 'contentId' in ruleInput) {
```

Comment:
Seems like a regression caused by https://github.com/oppia/oppia/commit/f4a6891b80c4e0322d5751243d2de05ccdbf9481

Why did we change this?

Reply:
![Screenshot from 2022-07-11 15-37-38](https://user-images.githubusercontent.com/77006897/178241252-c84fa6b5-d8be-4f6b-81e5-1a514f6c6734.png)


without this i am getting type error.  `Property 'contentId' does not exist on type 'InteractionRuleInputs'.
 Property 'contentId' does not exist on type 'AlgebraicExpressionRuleInputs'` and jeevesh garg is going to fix the type error because interaction type needs to be corrected at the root level. meanwhile, I am using this logic to solve the problem.